### PR TITLE
[HGCal] Updates on HGCalValidator 

### PR DESCRIPTION
**Pull request by apsallid**
#### PR description:

We update (@rovere , @felicepantaleo) the HGCalValidator package, anticipating the full deployment of TICL. In more detail: 

1. The package now can run through multiple input objects. The code was also restructured to avoid label dependent if statements that will lead to crash/problematic runs.
2. Removed contiguous and non contiguous histos from score. 
3. Removed cell association plots from multicluster section. 
4. Plotting script was updated removing the abovementioned plots. 
5. Work was rebased on top of the huge PR #28092. 

#### PR validation:

We tested the PR with TICL and without TICL customization. 

#### if this PR is a backport please specify the original PR:

This is not a backport.

### CODE REVIEW

**Comment by apsallid** [comment]
For some checks I accidentally committed Validation/RecoTrack/python/plotting/validation.py which I later added as it is. If this is a problem I can close this PR and open a new one with just one commit of all the changed files.

File: Validation/HGCalValidation/python/PostProcessorHGCAL_cfi.py [hunk 1]
```diff
@@ -24,22 +24,18 @@
 
 eff_multiclusters = ["effic_eta 'MultiCluster Efficiency vs #eta' Num_CaloParticle_Eta Denom_CaloParticle_Eta"]
 eff_multiclusters.extend(["effic_phi 'MultiCluster Efficiency vs #phi' Num_CaloParticle_Phi Denom_CaloParticle_Phi"])
-eff_multiclusters.extend(["duplicate_eta 'MultiCluster Duplicate(Split) Rate vs #eta' NumDup_MultiCluster_Eta Denom_MultiCluster_Eta"])
-eff_multiclusters.extend(["duplicate_phi 'MultiCluster Duplicate(Split) Rate vs #phi' NumDup_MultiCluster_Phi Denom_MultiCluster_Phi"])
+eff_multiclusters.extend(["duplicate_eta 'MultiCluster Duplicate(Split) Rate vs #eta ' NumDup_MultiCluster_Eta Denom_MultiCluster_Eta"])
+eff_multiclusters.extend(["duplicate_phi 'MultiCluster Duplicate(Split) Rate vs #phi ' NumDup_MultiCluster_Phi Denom_MultiCluster_Phi"])
 eff_multiclusters.extend(["fake_eta 'MultiCluster Fake Rate vs #eta' Num_MultiCluster_Eta Denom_MultiCluster_Eta fake"])
-eff_multiclusters.extend(["fake_phi 'MultiCluster Fake Rate vs #phi' Num_MultiCluster_Phi Denom_MultiCluster_Phi fake"])
+eff_multiclusters.extend(["fake_phi 'MultiCluster Fake Rate vs #phi  Num_MultiCluster_Phi Denom_MultiCluster_Phi fake"])
```
Comment (jfernan2) [inline Validation/HGCalValidation/python/PostProcessorHGCAL_cfi.py]: Not sure why the ' is removed here in this case...

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test workflow 20493.52

File: Validation/HGCalValidation/python/PostProcessorHGCAL_cfi.py (hunk 1 above)
Comment (rovere) [reply Validation/HGCalValidation/python/PostProcessorHGCAL_cfi.py]: @jfernan2 thanks for spotting: that's most likely  unless the `DQMGenericClient` has a really good and smart parser...

Comment (apsallid) [reply Validation/HGCalValidation/python/PostProcessorHGCAL_cfi.py]: @jfernan2 @rovere The honest truth is that I had also seen that after checking that the relevant plot is not produced but then a couple of emails arrived and I diverted. I am getting old. Let me correct this and push a commit.

**Comment by perrotta** [comment]
@cmsbuild please test workflow 20493.52

**Comment by jfernan2** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by perrotta** [comment]
+1
- The only modification in reco is in RecoHGCal/TICL/python/ticl_iterations.py, and it is consequent to the updates integrated here in the ```hgcalValidator``` module
- Jenkins tests pass: changes are visible in the HGCal DQM tests outputs, nothing in reco

**Comment by fabiocos** [comment]
+1

no impact observed in memory usage

**Pull request merged**